js/stat_test.ts

```diff
@@ -1,40 +1,65 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
 import { testPerm, assert, assertEqual } from "./test_util.ts";
 
-// TODO Add tests for modified, accessed, and created fields once there is a way
-// to create temp files.
-testPerm({ read: true }, async function statSyncSuccess() {
-  const packageInfo = Deno.statSync("package.json");
+const isWindows = Deno.platform.os === "win";
+
+async function success(
+  func: ((string) => Deno.FileInfo) | ((string) => Promise<Deno.FileInfo>)
+): Promise<void> {
+  const packageInfo = await func("package.json");
   assert(packageInfo.isFile());
+  assert(!packageInfo.isDirectory());
   assert(!packageInfo.isSymlink());
+  assertEqual(packageInfo.name, "package.json");
+  assertEqual(packageInfo.path, "package.json");
 
-  const testingInfo = Deno.statSync("testing");
+  const testingInfo = await func("testing");
+  assert(!testingInfo.isFile());
   assert(testingInfo.isDirectory());
   assert(!testingInfo.isSymlink());
+  assertEqual(testingInfo.name, "testing");
+  assertEqual(testingInfo.path, "testing");
 
-  const srcInfo = Deno.statSync("src");
+  const srcInfo = await func("src");
+  assert(!srcInfo.isFile());
   assert(srcInfo.isDirectory());
   assert(!srcInfo.isSymlink());
-});
+  console.log(`\nNAME:" '${srcInfo.name}'`);
+  console.log(`\nPATH: '${srcInfo.path}'`);
+  assertEqual(srcInfo.name, "src");
```

Comment:
to inline the CI (and local) error:

```
test statSyncSuccess_permR1W0N0E0U0 
NAME:" 'src'
PATH: 'src'... FAILED
Error
    at assert (/home/travis/build/denoland/deno/js/deps/https/deno.land/x/std/testing/mod.ts:6:19)
    at success (/home/travis/build/denoland/deno/js/stat_test.ts:27:5)
    at async statSyncSuccess (/home/travis/build/denoland/deno/js/stat_test.ts:62:5)
    at async runTests (/home/travis/build/denoland/deno/js/deps/https/deno.land/x/std/testing/mod.ts:202:13)
```

Reply:
Oh, the line numbers are incorrect! (Should've done some better print statements).
This was my mistake.